common/lib/xmodule/xmodule/modulestore/draft_and_published.py

```diff
@@ -134,6 +138,59 @@ def _flag_publish_event(self, course_key):
                 # We remove the branch, because publishing always means copying from draft to published
                 self.signal_handler.send("course_published", course_key=course_key.for_branch(None))
 
+    def remove_update_item_parent(self, item_location, new_parent_location, old_parent_location, user_id,
+                                  insert_at=None):
+        """
+        Removes block reference for old parent and attacehs to new parent.
+
+        Arguments:
+            item_location (BlockUsageLocator)    : Locator of item.
+            new_parent_location (BlockUsageLocator)  : New parent block locator.
+            old_parent_location (BlockUsageLocator)  : Old parent block locator.
+            insert_at (int) : Insert item at the particular index in new parent.
+            user_id (int)   : User id
+
+        Returns:
+           location or None:    source item location if updated otherwise None.
+        """
+        try:
+            source_item = self.get_item(item_location)
+            old_parent_item = self.get_item(old_parent_location)
+            new_parent_item = self.get_item(new_parent_location)
+        except ItemNotFoundError:
+            log.error('Unable to find the item %s', unicode(item_location))
```

Comment:
The exception here could be triggered by item_location, old_parent_location, or new_parent_location. The logged error has a 67% chance of being misleading.